### PR TITLE
Fixes 2941: Add RpmRepositoryVersionErrataList to tangy

### PR DESCRIPTION
**Pull request by Andrewgdewar**
Adds a function to list errata in repository versions, supports pagination and filtering on errata id, type, and severity

Can be tested with this [backend PR](https://github.com/content-services/content-sources-backend/pull/608)

### CODE REVIEW

File: pkg/tangy/rpm.go [hunk 1]
```diff
@@ -216,6 +234,75 @@ func (t *tangyImpl) RpmRepositoryVersionEnvironmentSearch(ctx context.Context, h
...
+	if filterOpts.ID != "" {
+		filterBytes.WriteString(" AND re.id ILIKE CONCAT( '%', @idFilter::text, '%')")
+	}
+	if filterOpts.Type != "" {
+		filterBytes.WriteString(" AND re.type = @typeFilter::text")
+	}
+	if filterOpts.Severity != "" {
+		filterBytes.WriteString(" AND re.severity = @severityFilter::text")
+	}
+	filterQuery := filterBytes.String()
```
Comment (Andrewgdewar) [inline pkg/tangy/rpm.go]: @jlsherrill  I'm not sure this is needed, I'm not super familiar with pgx. 
Also I would expect the AND re.severity = @severityFilter::text to be an "equal" not "like" comparison. 
But that is what I am seeing.

Comment (rverdile) [reply pkg/tangy/rpm.go]: Assuming this is working I think it's okay to concat strings like this. Although it looks like filterBytes is a `bytes.Buffer` then you're converting it to a string. You should probably just make the filter a string.

Comment (rverdile) [reply pkg/tangy/rpm.go]: I'm not sure what you mean about how the severity filter should be an "equal" and not a "like" filter. It looks like it is an "equal" filter?

Comment (Andrewgdewar) [reply pkg/tangy/rpm.go]: The "bytes.Buffer" method was recommended as best practice upon research of concatonating strings (I also felt it was weird but when in Golang...)

For the filter, I expected it to work like "Where X = Y" not "Where X LIKE Y" as in partial match to the query is permitted (IE "the" => "theme" should not work with "=" or so my understanding went.

Comment (jlsherrill) [reply pkg/tangy/rpm.go]: i assume byte.Buffer is more efficient when concatenating a lot of strings, but i think for our purposes, its likely not actually much difference (as we're just concatenating ~10 strings at most)

File: internal/test/integration/rpm_test.go [hunk 2]
```diff
@@ -26,6 +26,23 @@ type RpmSuite struct {
...
+
+	repoHref, remoteHref, err := r.client.CreateRepository(r.domainName, testRepoNameWithErrata, testRepoWithErrata)
+	require.NoError(t, err)
+
+	syncTask, err := r.client.SyncRpmRepository(repoHref, remoteHref)
+	require.NoError(t, err)
+
+	_, err = r.client.PollTask(syncTask)
+	require.NoError(t, err)
+}
```
Comment (rverdile) [inline internal/test/integration/rpm_test.go]: Could you refactor this to update the existing "CreateTestRepository" method to take a name and URL as input? Then we can use that method for any repository creation.

File: internal/test/integration/rpm_test.go [hunk 3]
```diff
@@ -244,6 +262,49 @@ func (r *RpmSuite) TestRpmRepositoryVersionEnvironmentSearch() {
 	assert.Len(r.T(), search, 0)
 }
 
+func (r *RpmSuite) TestRpmRepositoryVersionErrataListFilter() {
```
Comment (rverdile) [inline internal/test/integration/rpm_test.go:249]: Could you also add some cases to make sure that limit and offset are working?

**Comment by xbhouse** [comment]
i haven't found a repo with packages that include severities in the errata, so i haven't been able to really test that filter yet here. i've verified it works with the backend and UI, but if you know of any repos i could use to test here let me know

**Comment by swadeley** [comment]
> i haven't found a repo with packages that include severities in the errata, so i haven't been able to really test that filter yet here. i've verified it works with the backend and UI, but if you know of any repos i could use to test here let me know

Hi, this repo has severity:
https://stephenw.fedorapeople.org/fakerepos/multiple_errata/

**Comment by xbhouse** [comment]
@swadeley do the package errata in that repo have an updatedDate? trying to view that repo's package errata has surfaced an error: `can't scan into dest[6]: cannot scan NULL into *string"`

these are the values expected and i'm just assuming updatedDate is missing since it's the 7th one:
`id, errataId, title, summary, description, issuedDate, updatedDate, type, severity, rebootSuggested`

if updatedDate can be null (which makes sense if it can), i'll need to add a change

**Comment by xbhouse** [comment]
ok, changed updatedDate to *string and can now view the errata. thanks again

**Review by rverdile — APPROVED** [review]
looks good! remember to do the release after you merge!

**Pull request merged**